source/utils/paragraphHelper.py

```diff
@@ -0,0 +1,180 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2022 NV Access Limited, Rob Meredith
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+import api
+import speech
+import controlTypes
+import textInfos
+import tones
+from NVDAObjects.IAccessible.winword import WordDocument as IAccessibleWordDocument
+from NVDAObjects.UIA.wordDocument import WordDocument as UIAWordDocument
+
+MAX_LINES = 250  # give up after searching this many lines
+
+
+def getTextInfoAtCaret() -> textInfos.TextInfo:
+	# returns None if not editable text or document
```

Comment:
If Word/Outlook is blocked, we can postpone that fix to a future PR/issue.
Due to the performance issues, I think it would be good to disable Word until this is fixed.

```suggestion
	# returns None if 
	# - not editable text or;
	# - in a Word document, as Word has performance issues
```